odaridavid/POLITICO-API
# #163874733 Fixes Delete Same Id More Than Once

### What Does This PR do?
- Adds Test for deleting a party twice with the expectation of a 400 Bad Request
status Code
- Refactors Remove item in Model Class to only check id
- Refactors function generating a response to cater for non-existent ids
by returning a 400 Bad Request code

### Description of Task To Be Completed?

Have the `DELETE` Method Request on the `/parties` Endpoint working for deleting the same item twice edge case and returning a JSON response

### How Can This Be Manually Tested 
1. Clone The Repository and git checkout to bg-fix-delete-163874733.
2. Start Flask Virtual environment and export configuration details to run an instance of flask app.
3. On Postman or Similar API, Testing Client. Use the `POST` Method to add a party in the format below to the `/parties` endpoint.
{
	"name":"Party Name",
	"hqAddress":"Party Address",
	"logoUrl":"Party Logo Url"
}
4. Check Id generated and use the `DELETE` method on the `/parties/(generated_id)` twice it should return a 400 Response code and "Item Not Found" error message

### Relevant PT Stories

_#163874733_

### Screenshots 

![screenshot from 2019-02-11 20-36-24](https://user-images.githubusercontent.com/20339228/52582513-6a6eb100-2e3e-11e9-9721-858630c24754.png)
